app/facade.py

```diff
@@ -0,0 +1,259 @@
+from aries_cloudcontroller import AriesAgentController, AriesTenantController
+from fastapi import Header, HTTPException
+from contextlib import asynccontextmanager
+import json
+import requests
+import os
+import logging
+
+admin_url = os.getenv("ACAPY_ADMIN_URL")
+admin_port = os.getenv("ACAPY_ADMIN_PORT")
+is_multitenant = os.getenv("IS_MULTITENANT", False)
+
+
+logger = logging.getLogger(__name__)
+
+
+@asynccontextmanager
+async def create_controller(req_header: Header):
+    """
+    Instantiate an AriesAgentController or a TenantController
+    based on header attributes
+
+    Parameters:
+    -----------
+    req_header: Header
+        The header object containing (wallet_id, jwt_token) or api_key
+
+    Returns:
+    --------
+    controller: aries_cloudcontroller instance
+        A generator of AriesAgentController or TenantController object
+        (for use in contextmanager)
+    """
+    is_valid_tenant_header = "wallet_id" in req_header and "tenant_jwt" in req_header
+    is_valid_admin_header = "api_key" in req_header
+    is_valid_header = req_header and (is_valid_tenant_header or is_valid_admin_header)
+    if is_valid_header:
+        req_header = eval(req_header)
+        if "api_key" in req_header:
+            controller = AriesAgentController(
+                admin_url=f"{admin_url}:{admin_port}",
+                api_key=req_header["api_key"],
+                is_multitenant=is_multitenant,
+            )
+        else:
+            controller = AriesTenantController(
+                admin_url=f"{admin_url}:{admin_port}",
+                wallet_id=req_header["wallet_id"],
+                tenant_jwt=req_header["tenant_jwt"],
+            )
+    else:
+        raise HTTPException(
+            status_code=400,
+            detail="Bad headers. Either provide an api_key or both wallet_id and tenant_jwt",
+        )
+    try:
+        yield controller
+    except Exception as e:
+        # We can only log this here and not raise an HTTPExeption as
+        # we are past the yield. See here: https://fastapi.tiangolo.com/tutorial/dependencies/dependencies-with-yield/#dependencies-with-yield-and-httpexception
+        logger.error(f"{e!r}")
+    finally:
+        await controller.terminate()
+
+
+async def create_did(controller):
```

Comment:
I'd be keen to start adding type annotations to these kinds of methods... will help the dev who uses it to know what they're dealing with (And in fact ide's can utilise that to help with code insight) - not necessary for this review cycle, just flagging it for future